ros/src/can_proc/scripts/thrust_proc.py

```diff
@@ -15,16 +25,17 @@ def message_received(msg):
   esc_pub = rospy.Publisher('esc_single', esc_single_msg,
       queue_size= 10)
 
-  sub = rospy.Subscriber('thrust_control', thrust_control_msg,
+  sub = rospy.Subscriber('final_thrust', final_thrust_msg,
       message_received)
 
   rate = rospy.Rate(10) # 10hz
   # TODO: I2C related activities
   while not rospy.is_shutdown():
```

Comment:
ROS is built to be an event driven architecture. Which means when it receives a message, it'll spawn an interrupt and start the Subscriber callback function almost immediately. This gives us much faster response times than cyclic sleep rating does. Put this code to publish CAN messages in the message_received() function so it publishes the translated messages as soon as it receives a message.

So instead of a while loop with rospy.is_shutdown(). It'll just be a single call to rospy.spin() to pause the main thread.